### PR TITLE
Add source www.Allocine.fr

### PR DESCRIPTION
**Pull request by DeadlyJesus**
### Type of Change
New Feature: New Parser

### Description of the Change
Adds a parser for the movie rating website www.Allocine.fr

### Verification Process
Ran it on my own account (~700 reviews) and a test account made to test special cases (eg. Movie details page is missing release year / French release year != International release year)

### CODE REVIEW

**Comment by StegSchreck** [comment]
Hey @DeadlyJesus 
Thank you very much for your effort! In order to make the PR complete, I would ask you to write unit tests for the new parser and place them at `tests/unit/allocine/test_allocine_ratings_parser.py`. 
Additionally, it would good to adapt the content of `README.md` and `credentials.cfg.orig` to make it easier for others to use the new functionality.

File: RatS/allocine/allocine_ratings_parser.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    @staticmethod
+    def _get_movie_year(movie_details_page, movie_id):
+        year = movie_details_page.find(class_='date')
+        if year:
+            return int(year.text[-4:])
+        else:
+            return 0
+
+    @staticmethod
+    def _get_movie_original_title(movie_details_page, movie_id):
```
Comment (StegSchreck) [inline RatS/allocine/allocine_ratings_parser.py]: `movie_id` is unused here, it can be removed

File: RatS/allocine/allocine_ratings_parser.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        movie['year'] = self._get_movie_year(movie_details_page, movie[self.site.site_name.lower()]['id'])
+        # Use original movie title instead of french one
+        movie['title'] = self._get_movie_original_title(movie_details_page,
+                                                        movie[self.site.site_name.lower()]['id'])
+        if self.site.site_name.lower() not in movie:
+            movie[self.site.site_name.lower()] = dict()
+        movie[self.site.site_name.lower()]['my_rating'] = rating
+
+    @staticmethod
+    def _get_movie_year(movie_details_page, movie_id):
```
Comment (StegSchreck) [inline RatS/allocine/allocine_ratings_parser.py]: movie_id is unused here, it can be removed

File: RatS/allocine/allocine_ratings_parser.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    @staticmethod
+    def _get_movie_original_title(movie_details_page, movie_id):
+        original_title = movie_details_page.find('span', class_='what', text=' Titre original ')
+        if original_title:
+            return original_title.parent.find(class_='that').text
+        else:
+            return movie_details_page.find('div', class_='titlebar-title-lg').text
+
+    @staticmethod
+    def _get_movie_my_rating(movie_details_page, movie_id):
```
Comment (StegSchreck) [inline RatS/allocine/allocine_ratings_parser.py]: movie_id is unused here, it can be removed

File: RatS/allocine/allocine_ratings_parser.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+        return pages_count
+
+    @staticmethod
+    def _get_movie_tiles(movie_listing_page):
+        return movie_listing_page.find_all('div', class_='card')
+
+    @staticmethod
+    def _get_movie_title(movie_tile):
+        # Original movie title is found later on the movie details page
```
Comment (StegSchreck) [inline RatS/allocine/allocine_ratings_parser.py]: Suggestion for improving the clarification comment:
`At this point we only see the French movie title. Therefore, the original movie title is parsed later on the movie details page using _get_movie_original_title`

File: RatS/allocine/allocine_ratings_parser.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
+from bs4 import BeautifulSoup
+
+from RatS.base.base_ratings_parser import RatingsParser
+from RatS.allocine.allocine_site import Allocine
+
+
+class AllocineRatingsParser(RatingsParser):
```
Comment (StegSchreck) [inline RatS/allocine/allocine_ratings_parser.py]: Please rename to `AlloCineRatingsParser`

File: RatS/allocine/allocine_site.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import time
+
+from selenium.common.exceptions import NoSuchElementException
+
+from RatS.base.base_site import Site
+
+
+class Allocine(Site):
```
Comment (StegSchreck) [inline RatS/allocine/allocine_site.py]: Please rename to `AlloCine`

**Comment by StegSchreck** [comment]
This could be the new overview picture for the `README.md`. You can use this to overwrite `RatS/img/RatS_Ensemble.png`

image

**Comment by StegSchreck** [comment]
Hey @DeadlyJesus, are you making progress on this or do you need help?

**Comment by DeadlyJesus** [comment]
Hi @StegSchreck, I've done all the minor modification you asked, but I haven't found the time to work on the unit test during the holidays. I will do that as soon as possible.

**Comment by StegSchreck** [comment]
Hey @DeadlyJesus.
Are making any progress on this PR? You mentioned you have made some improvements, but didn't update the PR accordingly. If you find the time, I would be happy to see the completion of the PR.

**Pull request closed without merging**